### PR TITLE
Add support for parameter `rename` on ipahostgroup.

### PR DESCRIPTION
**Pull request by rjeffman**
FreeIPA 4.8.7 introduced an option to rename an existing hostgroup.
This patch adds support for renaming hostgroups if the option is
available on installed IPA version.

### CODE REVIEW

**Comment by rjeffman** [comment]
This patch has been tested against Fedora 32 with with `updates-testing` and the following RPM versions:
* freeipa-selinux-4.8.7-1.fc32.noarch
* freeipa-client-common-4.8.7-1.fc32.noarch
* freeipa-server-4.8.7-1.fc32.x86_64
* freeipa-healthcheck-core-0.5-3.fc32.noarch
* freeipa-common-4.8.7-1.fc32.noarch
* freeipa-client-4.8.7-1.fc32.x86_64
* freeipa-server-common-4.8.7-1.fc32.noarch
* freeipa-server-dns-4.8.7-1.fc32.noarch

**Comment by rjeffman** [comment]
I'm marking this PR as `blocked`, until the required version is available on a `stable` Fedora repo.

**Comment by t-woerner** [comment]
I would more like to have a new state `renamed`. State `present` with `rename` feels strange.

**Comment by t-woerner** [comment]
A rename and also further changes at the same time is unexpected. The rename should be something special as it can not be idempotent.

If a rename and want to have other changes should be done at the same time... what will be the result
1) If there is a group to rename
2) If the group has been renamed already
3) If there is no such group
4) If the target group already exists altogether with the source group

**Comment by rjeffman** [comment]
Added state `renamed` to the module to handle hostgroup renaming.

File: tests/hostgroup/test_hostgroup_rename.yml [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+  - name: Ensure host-group databases is present, again
+    ipahostgroup:
+      ipaadmin_password: SomeADMINpassword
+      name: databases
+      state: present
+    register: result
+    failed_when: result.changed
+
+  - name : Rename host-group from `databases` to `datalake`
```
Comment (t-woerner) [inline tests/hostgroup/test_hostgroup_rename.yml:33]: There is a space before ":".

File: tests/hostgroup/test_hostgroup_rename.yml [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    ipahostgroup:
+      ipaadmin_password: SomeADMINpassword
+      name: databases
+      rename: datalake
+      state: renamed
+    register: result
+    failed_when: not result.changed
+
+  - name : Rename host-group from `databases` to `datalake`, again
```
Comment (t-woerner) [inline tests/hostgroup/test_hostgroup_rename.yml:42]: There is a space before ":".

File: tests/hostgroup/test_hostgroup_rename.yml [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    ipahostgroup:
+      ipaadmin_password: SomeADMINpassword
+      name: databases
+      rename: datalake
+      state: renamed
+    register: result
+    failed_when: result.changed
+
+  - name : Rename host-group with same name.
```
Comment (t-woerner) [inline tests/hostgroup/test_hostgroup_rename.yml:51]: There is a space before ":".

File: tests/hostgroup/test_hostgroup_rename.yml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    ipahostgroup:
+      ipaadmin_password: SomeADMINpassword
+      name: datalake
+      rename: datalake
+      state: renamed
+    register: result
+    failed_when: result.changed
+
+  - name : Rename inexistent host-group.
```
Comment (t-woerner) [inline tests/hostgroup/test_hostgroup_rename.yml:60]: There is a space before ":".

**Comment by t-woerner** [comment]
Should a repeated use of `action: renamed` make sure that a hostgroup with the new name exists additionally?

File: tests/hostgroup/test_hostgroup_rename.yml (hunk 1 above)
Comment (rjeffman) [reply tests/hostgroup/test_hostgroup_rename.yml:33]: fixed.

File: tests/hostgroup/test_hostgroup_rename.yml (hunk 2 above)
Comment (rjeffman) [reply tests/hostgroup/test_hostgroup_rename.yml:42]: fixed

File: tests/hostgroup/test_hostgroup_rename.yml (hunk 4 above)
Comment (rjeffman) [reply tests/hostgroup/test_hostgroup_rename.yml:60]: fixed

File: tests/hostgroup/test_hostgroup_rename.yml (hunk 3 above)
Comment (rjeffman) [reply tests/hostgroup/test_hostgroup_rename.yml:51]: fixed

**Comment by rjeffman** [comment]
The current implemented behavior is:
* Rename if `name` exists and `rename` doesn't. (result.changed is True)
* Do nothing if `name` does not exist and `rename` does, or if `name` equals to `rename`. (result.changed is False)
* Fail if neither or both `name` and `rename` exist (result.failed is True).

**Comment by t-woerner** [comment]
The test failure is due to not supporting the rename in IPA versions prior to 4.7.8. Therefore this error needs to be ignored here.

**Pull request merged**